lib/player.js

```diff
@@ -5725,16 +5725,20 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     if (this.loadMode_ == shaka.Player.LoadMode.MEDIA_SOURCE) {
       // Filter by current videoId and has audio.
       filteredTracks = variants.filter((t) => {
-        return t.originalVideoId === active.originalVideoId && t.audioCodec;
+        return t.originalVideoId === active.originalVideoId &&
+            t.videoId === active.videoId && t.audioCodec;
```

Comment:
isn't videoId enough?

Reply:
There are some special cases of HLS that do not :(